src/script/browser/CheckBrowser.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {QUERY_KEY} from '../auth/route';
+
+const isMobile = (): boolean => 'ontouchstart' in window || !!navigator.maxTouchPoints;
+
+const isOauth = (): boolean => location?.hash?.includes(QUERY_KEY.SCOPE) ?? false;
+
+const supportsCookies = (): boolean => navigator.cookieEnabled;
+/* return new Promise((resolve, reject) => {
+     switch (navigator.cookieEnabled) {
+       case true:
+         return resolve();
+       case false:
+         return reject(new Error());
+       default:
+         Cookies.set(cookieName, 'yes');
+         if (Cookies.get(cookieName)) {
+           Cookies.remove(cookieName);
+           return resolve();
+         }
+         return reject(new Error());
+     }
+   });*/
+
+const supportsIndexDB = (): boolean => 'indexedDB' in window;
+/*{
```

Comment:
Same. I'm not sure if we have to check the IndexDB for Firefox in this complex way

Reply:
we do need such a check, I'm afraid. 

If you are afraid the async will not work, it's going to be okay since it's going to be safe guarded by the initial `isBrowserSupported` call. 

So you can go


```
if (isBrowserSupported()) {
  checkIndexedDBSupport().then((isSupported) => {if (!isSupported) { redirect() }});
}
```

Reply:
done